pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -1248,6 +1238,57 @@ func (p *Planner) ensureRKEStateSecret(controlPlane *rkev1.RKEControlPlane, full
 	}, nil
 }
 
+func getIPAddresses(secrets corecontrollers.SecretCache, machine *capi.Machine) (string, string) {
+	internalIPAddress, internalAddressProvided := machine.Annotations[InternalAddressAnnotation]
+	ipAddress, addressProvided := machine.Annotations[AddressAnnotation]
+
+	// Unmanaged machines won't have a state secret
+	if machine.Annotations["objectset.rio.cattle.io/id"] != "unmanaged-machine" && (!internalAddressProvided || !addressProvided) {
```

Comment:
The check for "unmanaged-machine" I believe is redundant because in the block of code you have you are optmistically looking for the secret and only processing it if found, so you can just.  If you really want a check for the type, the more proper check would be that apiVersion == "rke-machine.cattle.io/v1".  But I'd just remove the check.

Reply:
Actually, I believe  the `(!internalAddressProvided || !addressProvided)` is not needed.  boolean logic always confuses me, but I believe since below you are already gating on `if !foo` you don't need this check.

Reply:
Done.